agent-inject/agent/container_env.go

```diff
@@ -42,6 +44,21 @@ func (a *Agent) ContainerEnvVars(init bool) ([]corev1.EnvVar, error) {
 			Name:  "VAULT_CONFIG",
 			Value: b64Config,
 		})
+
+		// Add IRSA AWS Env variables for vault containers
+		if a.Pod != nil {
+			envMap := a.getEnvsFromContainer(a.Pod)
+			if len(envMap) == 0 || len(envMap) >= 2 {
+				for k, v := range envMap {
+					envs = append(envs, corev1.EnvVar{
+						Name:  k,
+						Value: v,
+					})
+				}
+			} else {
+				log.Println("WARN: Could not find 'AWS ROLE/ AWS_WEB_IDENTITY_TOKEN_FILE' env variables")
+			}
```

Comment:
I think something like this would be simpler, and we probably don't want to log every time the AWS env vars aren't found:

```suggestion
			for k, v := range envMap {
				envs = append(envs, corev1.EnvVar{
					Name:  k,
					Value: v,
				})
			}
```

If we want to keep the log, I'd suggest changing it to trace or debug level, and it'll need to be a logger based off the hclog in handler.